### PR TITLE
Use experiment in ert-storage for ert3

### LINKED ISSUE #1554: Make ert3 use experiments in ert-storage
The concept of experiments is now supported in ert-storage. This included breaking changes in ert-storage api. ert3 should convert the usage of ert-storage to the new api.

### PR DESCRIPTION
**Pull request by ManInFez**
**Issue**
Resolves #1554

### CODE REVIEW

File: ert3/storage/_storage.py [hunk 1]
```diff
@@ -1,109 +1,84 @@
...
-def _generate_storage_location(workspace: Union[str, Path]) -> Path:
-    workspace = Path(workspace)
-    return workspace / ert3._WORKSPACE_DATA_ROOT / _STORAGE_FILE
-
-
-def _assert_storage_initialized(storage_location: Path) -> None:
-    if not os.path.isfile(storage_location):
-        raise ValueError("Storage is not initialized")
+def _get_experiment_by_name(experiment_name: str) -> Mapping[str, Any]:
+    response = requests.get(url=f"{_STORAGE_URL}/experiments")
```
Comment (dotfloat) [inline ert3/storage/_storage.py:20]: This should probably be done on the ert storage server side. Add a `params={"name": experiment_name}`, and have ERT Storage modify its SQL query accordingly.

File: ert3/storage/_storage.py [hunk 2]
```diff
@@ -1,109 +1,84 @@
...
+    assert response.status_code == 200
```
Comment (dotfloat) [inline ert3/storage/_storage.py:21]: Debateable, but asserts can be disabled by the user python-wide so don't use them in production code.

File: ert3/storage/_storage.py [hunk 3]
```diff
@@ -1,109 +1,84 @@
...
-
-    if os.path.exists(storage_location):
-        raise ValueError(f"Storage already initialized for workspace {workspace}")
-
-    if not os.path.exists(storage_location.parent):
-        os.makedirs(storage_location.parent)
-
-    with open(storage_location, "w") as f:
-        yaml.dump({}, f)
+    response = requests.get(url=f"{_STORAGE_URL}/experiments")
```
Comment (dotfloat) [inline ert3/storage/_storage.py:27]: Seems wasteful to get _all_ the experiments only to filter them client-side.

File: ert3/storage/_storage.py [hunk 4]
```diff
@@ -1,109 +1,84 @@
...
 def init_experiment(
     *, workspace: Union[str, Path], experiment_name: str, parameters: Iterable[str]
 ) -> None:
-    storage_location = _generate_storage_location(workspace)
-    _assert_storage_initialized(storage_location)
-
-    with open(storage_location) as f:
-        storage = yaml.safe_load(f)
+    if not experiment_name:
+        raise ValueError("Cannot initialize experiment with not name")
```
Comment (dotfloat) [inline ert3/storage/_storage.py:44]:
```suggestion
        raise ValueError("Cannot initialize experiment with no name")
```

File: ert3/storage/_storage.py (hunk 1 above)
Comment (ManInFez) [reply ert3/storage/_storage.py:20]: The should be discussed, but in my opinion ert3 should start to handle experiment ids and not rely on names, as experiments from different workspaces could share the same name, and workspace is not a thing in ert-storage.

**Review by dotfloat — APPROVED** [review]

**Pull request merged**